packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -337,12 +341,32 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
   TextEditingController get _effectiveController =>
       widget.controller ?? _controller!.value;
 
+  late AnimationController _animationController;
+  ScrollController? _ancestorScrollController;
+  double _maxHeight = 0.0;
+
   @override
   void initState() {
     super.initState();
     if (widget.controller == null) {
       _createLocalController();
     }
+     _animationController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 300),
+      value: 1.0,
+    );
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _calculateMaxHeight();
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _ancestorScrollController?.removeListener(_onScroll);
+    _ancestorScrollController = Scrollable.maybeOf(context)?.widget.controller;
```

Comment:
This is probably why it is only working on mobile.
ScrollableState.widget.controller will only have a controller if the user has provided one, or the ancestor ScrollView decided to assign the PrimaryScrollController.
See the _effectiveScrollController in ScrollableState, which accounts for this. It may need to be made public for this feature, although, you could use ScrollableState.position and listen to that alternatively. :)